File: OOPS_JAVA/LinkedList.java
package OOPS_JAVA;

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class LinkedList {

    public static class Node{
        int data;
        Node next;
    }

    public static class linkedlist{
        Node head;
        Node tail;
        int size;

        int size(){
            return size;
        }

        void display(){
            for (Node temp = this.head; temp != null ; temp = temp.next){
                System.out.print(temp.data + " --> ");
            }
            System.out.println();
        }
        void addLast(int val){ // last me add
            Node node = new Node(); // phle new node bnaoo
            node.data = val; // uske data me val daloo
            node.next = null; // uska next null pe kro bcz vo last hainn

            if (size == 0){ // agr ll empty h to head tail sbb vhi hoga
                head = tail = node;
            }
            else{ // tail ka next node pe kro(isse vo node add hoga)and tail ko node pe krke update krdo
                tail.next = node;
                tail = node;
            }
            size++;// size bdhaa do
        }
    }
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        LinkedList list = new LinkedList();
    
        String str = br.readLine();
        while(str.equals("quit") == false){
          if(str.startsWith("addLast")){
            int val = Integer.parseInt(str.split(" ")[1]);
            list.addLast(val);
          } else if(str.startsWith("size")){
            System.out.println(list.size());
          } else if(str.startsWith("display")){
            list.display();
          }
          str = br.readLine();
        }
    }
}

File: Linkedlist.py
class Node():
    def __init__(self):
        self.data = None
        self.next = None
        
class LinkedList():
    def __init__(self):
        self.head = None
        self.tail = None
        self.size = 0
        
    def length(self):
        print(self.size)
        return self.size

    def addfirst(self, data):
        node = Node()
        node.data = data
        if self.head == None:
            self.head = self.tail = node
            self.size+=1
        else:
            node.next = self.head
            self.head = node
            self.size +=1
    
    def addlast(self, data):
        node = Node()
        node.data = data
        if self.head == None:
            self.head = self.tail = node
            self.size+=1
        else:
            self.tail.next = node
            self.tail = node
            self.size +=1
            
    def addatany(self, idx, data):
        node = Node()
        node.data = data
        if idx >= 0 or idx < size():  
            if idx == 0:
                addfirst(data)
            elif idx == size() -1:
                addlast(data)
            else:
                pointer = 0
                temp = self.head
                while(pointer != idx-1): # ek phle rukenge
                    temp = temp.next
                node.next = temp.next
                temp.next = node
                
            
        
    def display(self):
        temp = self.head
        while(temp.next != None):
            print(temp.data, end = "-->")
            temp = temp.next
        print(temp.data, end = "")
        print()
        
ll =LinkedList()
ll.length()
ll.addfirst(10)
ll.addfirst(20)
ll.addfirst(30)
ll.display()
ll.length()
ll.addlast(50)
ll.display()
ll.length()